### PR TITLE
Example: Exposing styles as classes

### PR DESCRIPTION
**Pull request by MACSkeptic**
## Description

The idea is to expose Helix styles for consumers that are **not** using the helix ui toolkit nor webcomponents.

For that we need to expose every selector _globally_ as a class selector.

This is a proof of concept of exposing styles from helix components as pure css classes

I don't quite understand the way the helix css is put together, so I'm 100% sure there is a better way of doing this.

Screenshot 2020-11-22 at 18 51 46

example code:

```javascript
const App = () => {
  return (
    <>
      <div className="comparison">
        <fieldset>
          <legend>Pure HTML</legend>
          <div className="hx-busy" />
        </fieldset>
        <fieldset>
          <legend>Helix</legend>
          <hx-busy />
        </fieldset>
      </div>
      <div className="comparison">
        <fieldset>
          <legend>Pure HTML</legend>
          <div open className="hx-popover has-arrow" id="htmlPopover" position="bottom-right" style={{
            left: 0,
            position: 'relative'
          }}>
            <header>
              Popover Header
            </header>

            <div className="hx-div">
              <p>
                This is my popover body.
              </p>
            </div>

            <footer>
              <button className="hxBtn hxPrimary">Submit</button>
              <button className="hxBtn">Cancel</button>
            </footer>
          </div>
        </fieldset>
        <fieldset>
          <legend>Helix</legend>
          <hx-disclosure aria-controls="demoPopover" className="hxBtn">
            Toggle
          </hx-disclosure>
          <hx-popover id="demoPopover" position="bottom-right">
            <header>
              Popover Header
            </header>

            <hx-div>
              <p>
                This is my popover body.
              </p>
            </hx-div>

            <footer>
              <button className="hxBtn hxPrimary">Submit</button>
              <button className="hxBtn">Cancel</button>
            </footer>
          </hx-popover>
        </fieldset>
      </div>
    </>
  );
};
```

### CODE REVIEW

File: src/elements/hx-div/_shadow.scss [hunk 1]
```diff
@@ -16,7 +16,7 @@
       var(--hxPaddingBottom)
       var(--hxPaddingLeft);
 
-    #hxDiv {
+    .hx-div, #hxDiv {
```
Comment (MACSkeptic) [inline src/elements/hx-div/_shadow.scss:19]: basically we'd have to duplicate anything that is an id or tag matcher to also offer a class variant

File: src/scss/components/popover/_index.scss [hunk 2]
```diff
@@ -1,8 +1,10 @@
 @import "vars";
 @import "mixins";
 @import "../config";
+@import "../../base";
+@import "../../elements/_has-arrow";
```
Comment (MACSkeptic) [inline src/scss/components/popover/_index.scss:5]: I'm not sure if this imports should be here, maybe there is a better way of exposing this globally?

Comment (100stacks) [reply src/scss/components/popover/_index.scss:5]: Going this route, we would have a different implementation from the current HelixUI components.

**Review by 100stacks — COMMENTED** [review]
### @MACSkeptic, we have an experimental repo that may address this  - `helix-core`:

#### https://github.com/HelixDesignSystem/helix-core

It's far from complete.

**Review by 100stacks — CHANGES_REQUESTED** [review]
### POC PR #786.  Will reopen if needed.

**Pull request closed without merging**